apps/server/test/modules/task/submission.e2e-spec.ts

```diff
@@ -37,128 +37,250 @@ class API {
+			it('should return 401', async () => {
+				const taskId = 'id';
 
-			expect(status).toEqual(401);
-		});
-	});
+				const { status } = await api.findStatusesByTask(taskId);
 
-	describe('WHEN user is authorized, has permission', () => {
-		const setup = async () => {
-			const roles = roleFactory.buildList(1, {
-				permissions: [Permission.SUBMISSIONS_VIEW],
+				expect(status).toEqual(401);
 			});
-			const user = userFactory.buildWithId({ roles });
-			const task = taskFactory.buildWithId();
-			const courseGroup = courseGroupFactory.buildWithId();
-			const submission = submissionFactory.buildWithId({ task, student: user, grade: 97, courseGroup });
+		});
 
-			await em.persistAndFlush([submission]);
-			em.clear();
-			currentUser = mapUserToCurrentUser(user);
+		describe('WHEN user is authorized, has permission', () => {
+			const setup = async () => {
+				const roles = roleFactory.buildList(1, {
+					permissions: [Permission.SUBMISSIONS_VIEW],
+				});
+				const user = userFactory.buildWithId({ roles });
+				const task = taskFactory.buildWithId();
+				const courseGroup = courseGroupFactory.buildWithId();
+				const submission = submissionFactory.buildWithId({ task, student: user, grade: 97, courseGroup });
 
-			return { task, submission };
-		};
+				await em.persistAndFlush([submission]);
+				em.clear();
+				currentUser = mapUserToCurrentUser(user);
+
+				return { task, submission };
+			};
+
+			it('should return status', async () => {
+				const { task, submission } = await setup();
+
+				const { result: statuses } = await api.findStatusesByTask(task.id);
 
-		it('should return status', async () => {
-			const { task, submission } = await setup();
+				const expectedSubmissionStatuses = {
+					data: [
+						{
+							id: submission.id,
+							submitters: submission.getSubmitterIds(),
+							isSubmitted: submission.isSubmitted(),
+							isGraded: submission.isGraded(),
+							grade: submission.grade,
+							submittingCourseGroupName: submission.courseGroup?.name,
+						},
+					],
+				};
 
-			const { result: statuses } = await api.findStatusesByTask(task.id);
+				expect(statuses).toEqual(expectedSubmissionStatuses);
+			});
+		});
+
+		describe('with bad request data', () => {
+			it('should return status 400 for invalid taskId', async () => {
+				const response = await api.findStatusesByTask('123');
 
-			const expectedSubmissionStatuses = {
-				data: [
+				expect(response.status).toEqual(400);
+				expect(response.error.validationErrors).toEqual([
 					{
-						id: submission.id,
-						submitters: submission.getSubmitterIds(),
-						isSubmitted: submission.isSubmitted(),
-						isGraded: submission.isGraded(),
-						grade: submission.grade,
-						submittingCourseGroupName: submission.courseGroup?.name,
+						errors: ['taskId must be a mongodb id'],
+						field: 'taskId',
 					},
-				],
+				]);
+			});
+		});
+
+		describe('WHEN user is authorized and has no permission', () => {
```

Comment:
Accordingly to the above this should be "user is autheticated and does not have permission" or "user is authenticated but unauthorized" or just "user is unauthorized".